imapclient/notify_encode_test.go

```diff
@@ -0,0 +1,390 @@
+package imapclient
+
+import (
+	"bufio"
+	"bytes"
+	"testing"
+
+	"github.com/emersion/go-imap/v2"
+	"github.com/emersion/go-imap/v2/internal/imapwire"
+)
+
+func encodeToString(options *imap.NotifyOptions) (string, error) {
```

Comment:
Using table-driven tests should allow us to cut down the boiler plate here I believe: https://go.dev/wiki/TableDrivenTests